public/index.html

```diff
@@ -1,46 +1,243 @@
+				y  : (app.height / 2) - 50,
+				color: "green",
+				...sharedPlayerAttributes,
+			})
 
 			this.nodes.push({
-				id : 'black-box',
-				x  : 50,
-				y  : 0,
-				width  : 150,
-				height : 150,
-				color  : 'black'
-			});
+				id : "ball",
+				x  : (app.width / 2) - 10,
+				y  : (app.height / 2) - 10,
+				color  : "blue",
+				width  : 20,
+				height : 20,
+				speed  : 5,
+				dy : 0,
+				dx : 0,
+			})
+
+			// Define key bindings for movement
+			this.keydownBindings = {
+				"w": () => (this.getNode('player1').direction = -1),
+				"s": () => (this.getNode('player1').direction = 1),
+				"ArrowUp": () => (this.getNode('player2').direction = -1),
+				"ArrowDown": () => (this.getNode('player2').direction = 1),
+			}
+
+			this.keyupBindings = {
+				"w": () => (this.getNode('player1').direction = 0),
+				"s": () => (this.getNode('player1').direction = 0),
+				"ArrowUp": () => (this.getNode('player2').direction = 0),
+				"ArrowDown": () => (this.getNode('player2').direction = 0),
+			}
+
+			console.log("***** Press Space Bar to Start *****");
 		};
 
 		app.onUpdate = function(time){
-			this.getNode('black-box').y++;
+			player1Node = this.getNode('player1');
+			player2Node = this.getNode('player2');
+			ballNode = this.getNode('ball');
 
-			if(Math.cos(this.timestamp / 100) > 0){
-				this.getNode('red-box').direction = -1;
-			}else{
-				this.getNode('red-box').direction = 1;
-			}
+			// Control Player Movement
+			handlePlayerMovement(player1Node);
+			handlePlayerMovement(player2Node);
 
-			this.getNode('red-box').x+=this.getNode('red-box').direction;
+			// Handle Ball Movement and interractions
+			// Bounce off cieling and floor
+			if (ballNode.y <= 0 || ballNode.y >= (app.height - ballNode.height)) {
+				ballNode.dy *= -1;
+			}
+			// If Ball makes contact with Player1's paddle
+			if (ballNode.x <= (player1Node.x + player1Node.width) && 
+			ballNode.x >= player1Node.x && 
+			ballNode.y + (ballNode.height / 2) >= player1Node.y && ballNode.y + (ballNode.height / 2) <= player1Node.y + player1Node.height) {
+				modifyBallVelocity(ballNode, player1Node);
+			}	
+			// If Ball makes contact with Player2's paddle
+			else if (ballNode.x + ballNode.width >= player2Node.x && ballNode.x + ballNode.width <= player2Node.x + player2Node.width &&
+			ballNode.y + (ballNode.height / 2) >= player2Node.y && ballNode.y + (ballNode.height / 2) <= player2Node.y + player2Node.height) {
+				modifyBallVelocity(ballNode, player2Node);
+			}
+			else if (ballNode.x <= 0) {
+				console.log("PLAYER 2 SCORES!!");
+				app.p2Score++;
+				console.log(`****** SCORE *****\nPlayer 1: ${app.p1Score}\nPlayer 2: ${app.p2Score}\n*****************`);
+				this.nextRound();
+			} 
+			else if (ballNode.x >= (app.width - ballNode.width)) {
+				console.log("PLAYER 1 SCORES!!");
+				app.p1Score++;
+				console.log(`****** SCORE *****\nPlayer 1: ${app.p1Score}\nPlayer 2: ${app.p2Score}\n*****************`)
+				this.nextRound();
+			} 
```

Comment:
you've written some comments which are unnecessary if the code is modularized.

This should be broken into a few functions in a way that no comments are needed and everything is more easily understandable with the function names.

Reply:
Thank you, I have created functions to check if the ball is hitting a paddle as well as a function for checking if the ball is in contact with the floor or ceiling. I put those functions into a new utils.js. I also moved the other function definitions that I included into the utils.js file.